frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -38,6 +41,36 @@ function extractObjectWithMaxDepth(obj, depth = MAX_DEPTH) {
   };
 }
 
+/**
+ * Adds an opacity rating to each of the nodes based on where the node corresponding to
+ * the link is found.
+ * If no link is being hovered over, all nodes are fully visible
+ * If a link is being hovered over, then the corresponding node (if it exists) and all
+ * descendents are highlighted, everything else is greyed out
+ * @param {Object} data The data to traverse to add the opacity
+ * @param {string} link The link we are equating to, refers to the link being hovered on
+ * @param {float} currentOpacity Current opacity rating
+ */
+const addOpacity = (data, link, currentOpacity) => {
```

Comment:
Let's name this "changeOpacity" since there is no addition going on, just a change in the opacity value.

Reply:
How about `setOpacity` since the opacity is not necessarily changing? lmk if any issues with this